safehttp/plugins/fetch_metadata/fetch_metadata.go

```diff
@@ -0,0 +1,121 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package fetchmetadata
+
+import (
+	"github.com/google/go-safeweb/safehttp"
+)
+
+const (
+	enforceMode = "enforce"
+	reportMode  = "report"
+)
+
+// LoggingService TODO
+type LoggingService interface {
+	Log(ir *safehttp.IncomingRequest)
+}
+
+// Plugin implements Fetch Metadata functionality.
+// See https://www.w3.org/TR/fetch-metadata/ for more details.
+type Plugin struct {
+	mode   string
+	policy func(*safehttp.IncomingRequest) bool
+	logger LoggingService
+}
+
+// NewPlugin creates a new Fetch Metadata plugin using the defaultPolicy and
+// sets the mode to enforce.
+func NewPlugin() *Plugin {
+	return &Plugin{
+		mode:   enforceMode,
+		policy: defaultPolicy,
+	}
+}
+
+// defaultPolicy applies a Resource Isolation Policy to the
+// safehttp.IncomingRequest, allowing it to pass only if it conforms to it.
+//
+// See https://web.dev/fetch-metadata/ for more information.
+func defaultPolicy(r *safehttp.IncomingRequest) bool {
+	switch r.Header.Get("Sec-Fetch-Site") {
```

Comment:
Create a local variable `h := r.Header` to use instead of `r.Header`. Since `r.Header` is used so much this will improve readability